### PR TITLE
Added Substring StringBuilder functionality

### PR DESCRIPTION
**Pull request by varunu28**
**Changes**
 - Adding `Substring` functionality based upon discussions in https://github.com/linkdotnet/golang-stringbuilder/issues/5
 - Added tests to verify the functionality

### CODE REVIEW

File: stringbuilder.go [hunk 1]
```diff
@@ -289,6 +289,19 @@ func (s *StringBuilder) Reverse() *StringBuilder {
...
 }
 
+// Substring creates a substring from the string builder using a start and an end(exclusive) bound
+func (s *StringBuilder) Substring(start, end int) (string, error) {
+	if start < 0 {
+		return "", fmt.Errorf("start should always be greater than or equal to zero")
+	}
+	if end > s.position {
+		return "", fmt.Errorf("end cannot be greater than the length of string builder")
+	}
```
Comment (linkdotnet) [inline stringbuilder.go:299]: There should also be a check if `start < end`.

File: stringbuilder.go [hunk 2]
```diff
@@ -289,6 +289,19 @@ func (s *StringBuilder) Reverse() *StringBuilder {
 	return s
 }
 
+// Substring creates a substring from the string builder using a start and an end(exclusive) bound
```
Comment (linkdotnet) [inline stringbuilder.go]:
```suggestion
// Returns a substring from start (inclusive) to end (exclusive).
```

As the comment is above the method-name itself, I don't think we need the name again - plus it fits better to the overall pattern.

File: stringbuilder.go (hunk 1 above)
Comment (varunu28) [reply stringbuilder.go:299]: Ahh nice catch. Updated with additional check and a test

File: stringbuilder.go (hunk 2 above)
Comment (varunu28) [reply stringbuilder.go]: Done

**Review by linkdotnet — APPROVED** [review]

**Comment by linkdotnet** [comment]
Nice @varunu28  - will make a release right away.

**Pull request merged**